### PR TITLE
GUI: Support for default values in PacketParser

### PR DESCRIPTION
**Pull request by chayaRoten**
### PR Description

**Purpose:**
This Pull Request adds support for default values in the `PacketParser` class of the system. The goal is to enhance the data management capabilities of the `PacketParser` by allowing default values to be defined for different field types.

**Key Changes:**
1. **Addition of Field Types:**
   - New field types have been added to the function handler map, including `signed_int` and `boolean`.
   - Fields with default values can now be stored appropriately in the `PacketParser` data structure.

2. **Interface Improvements:**
   - When `PacketParser` loads JSON with fields, it now recognizes fields with default values and stores them correctly in the `defaultValues` map.

3. **Testing Enhancements:**
   - Tests have been added to ensure that default values are returned correctly for each field type, and in the case of an unknown field, a warning message is provided.

**Benefits:**
- This improvement increases the flexibility of the `PacketParser` and allows developers to set default values more easily, leading to reduced errors and a smoother operation of the system.

### CODE REVIEW

File: parser_json/json_example.json [hunk 1]
```diff
@@ -1,38 +1,45 @@
...
           "description": "The type of the field. Must be one of the following:",
           "allowed_values": [
-            "char_array",
-            "float_fixed",
-            "float_mantissa",
-            "double",
-            "signed_int",
-            "unsigned_int",
-            "bit_field",
+            "char_array",
```
Comment (ChedvaErblich) [inline parser_json/json_example.json]: Don't add those white spaces at the end of the lines

File: parser_json/json_example.json [hunk 2]
```diff
@@ -41,32 +48,38 @@
...
               "description": "The type of the sub-field. Must be one of the supported types.",
               "allowed_values": [
-                "char_array",
-                "float_fixed",
-                "float_mantissa",
-                "double",
-                "signed_int",
-                "unsigned_int",
-                "bit_field",
+                "char_array",
```
Comment (ChedvaErblich) [inline parser_json/json_example.json]: Same here.

File: parser_json/json_example.json [hunk 3]
```diff
@@ -1,38 +1,45 @@
...
           ],
           "example": "float_fixed"
         },
         "size": {
-          "description": "The size of the field in bits. Must be a multiple of 8, or 16, 32, 64, etc.",
+          "description": "The size of the field in bits. Must be a multiple of 8, or 16, ",
           "example": 32
         },
+        "default_value": {
+          "description": "Specifies the default value of the field. This value is optional.",
```
Comment (ChedvaErblich) [inline parser_json/json_example.json]: It's not clear that the 'default_value' entry in the JSON is optional. The explanation seems to suggest that the field must appear, but can be left unset. However, as we discussed, the code supports the option where the field doesn't exist at all. Please clarify this point.

File: parser_json/json_example.json [hunk 4]
```diff
@@ -41,32 +48,38 @@
...
                 "boolean"
               ],
               "example": "float_fixed"
             },
             "size": {
               "description": "The size of the sub-field in bits.",
               "example": 32
+            },
+            "default_value": {
+              "description": "Specifies the default value for the sub-field.",
```
Comment (ChedvaErblich) [inline parser_json/json_example.json]: Please mention here as well that it's optional to avoid any confusion for the user.
And in all other places,,,

**Review by ChedvaErblich — CHANGES_REQUESTED** [review]
Have you confirmed with the main control team that it works for them? 
Also, please inform the image processing team about the new code.

File: parser_json/json_example.json (hunk 4 above)
Comment (chayaRoten) [reply parser_json/json_example.json]: Done

File: parser_json/json_example.json (hunk 3 above)
Comment (chayaRoten) [reply parser_json/json_example.json]: Done

File: parser_json/json_example.json (hunk 2 above)
Comment (chayaRoten) [reply parser_json/json_example.json]: Done

File: parser_json/json_example.json (hunk 1 above)
Comment (chayaRoten) [reply parser_json/json_example.json]: Done

**Comment by chayaRoten** [comment]
Yes, the main control team reviewed the code and confirmed that it works for them. This code was developed in this way after I aligned positions with them.

**Pull request merged**